src/typegraph/rust/typegraph_macro/src/lib.rs

```diff
@@ -0,0 +1,105 @@
+mod types;
+
+use proc_macro::TokenStream;
+use quote::{quote, ToTokens};
+use syn::{
+    parse_macro_input as parse, punctuated::Punctuated, token::Comma, Expr, ExprArray, ExprLit,
+    ItemFn, Lit, LitBool, LitInt, LitStr, Meta, PatLit,
+};
+use types::{Cors, Rate, TypegraphInitParams};
+
+#[proc_macro_attribute]
+pub fn typegraph(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let input_fn = parse!(item as ItemFn);
+    let args = parse!(attr with Punctuated::<Meta, syn::Token![,]>::parse_terminated);
+
+    let mut tg_params = TypegraphInitParams::default();
+
+    for arg in args {
+        match arg {
+            Meta::NameValue(meta) => {
+                let ident = meta.path.require_ident().unwrap();
+                let expr = meta.value.to_token_stream().into();
+
+                match ident.to_string().as_str() {
+                    "name" => tg_params.name = parse!(expr as LitStr).value(),
+                    "path" => tg_params.path = parse!(expr as LitStr).value(),
+                    "prefix" => tg_params.prefix = parse!(expr as LitStr).value().into(),
+                    "dynamic" => tg_params.dynamic = parse!(expr as LitBool).value(),
```

Comment:
Curious to know what's your final decision on this? continue the proc macro approach or simplify like..
```rust
typegraph(params, |g: Graph| {
   g.expose(...);
});
```

Reply:
Actually, I plan to remove this and just add a `Drop` trait function where I'd put the CLI serialization operations.